src/steamship/utils/dict_mapper.py

```diff
@@ -0,0 +1,179 @@
+"""Helper classes for mapping third-party API response objects into Steamship model objects.
+
+Many third-party APIs produce data that is isomorphic, but differently structured, than Steamship plugin responses.
+These utility classes reduce the required conversion to a declarative specification, enabling faster, simpler, and more
+error-robust construction of Steamship Plugins which wrap third-party services.
+
+For a usage example, see the documentation on the `reshape_array_of_dicts` function below.
+"""
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Type, Union
+
+from pydantic import BaseModel
+
+from steamship import SteamshipError
+
+
+def get_value_at_keypath(
+    input_dict: Optional[Dict],
+    keypath: List[Union[str, int]],
+    expect_type: Optional[Any] = None,
+    required: bool = True,
+) -> Optional[Any]:
+    """Extracts value at the provided `keypath` from `dict`.
+
+    - If `expect_type` is not None, requires that any non-None return type match that type.
+    - if `throw_on_missing` is True (default), will throw if the return value is None
+    """
+    pointer = input_dict
+    traversed_keypath = []
+    for key in keypath:
+        if pointer is None:
+            if required:
+                raise SteamshipError(
+                    message=f"[get_keypath] After traversing keypath {traversed_keypath} in object, found null instead of container for expected key {key}."
+                )
+            return None
+
+        if isinstance(pointer, dict):
+            if key not in pointer:
+                if required:
+                    raise SteamshipError(
+                        message=f"[get_keypath] After traversing keypath {traversed_keypath} in object, unable to find expected key {key}."
+                    )
+                return None
+            traversed_keypath.append(key)
+            pointer = pointer.get(key, None)
```

Comment:
```suggestion
            pointer = pointer[key]
```

We can use pointer[key] here as we checked if key is present before.